src/EditorFeatures/Core/Organizing/OrganizeDocumentCommandHandler.cs

```diff
@@ -115,61 +93,110 @@ private static bool IsCommandSupported(EditorCommandArgs args, bool needsSemanti
             return false;
         }
 
-        public bool ExecuteCommand(SortImportsCommandArgs args, CommandExecutionContext context)
+        private bool ExecuteCommand(
+            EditorCommandArgs commandArgs,
+            CommandExecutionContext context,
+            Func<ITextSnapshot, IUIThreadOperationContext, Task<Document?>> getCurrentDocumentAsync,
+            Func<Document, CancellationToken, Task<Document>> getChangedDocumentAsync)
         {
-            using (context.OperationContext.AddScope(allowCancellation: true, EditorFeaturesResources.Organizing_document))
-            {
-                SortImports(args.SubjectBuffer, context.OperationContext);
-            }
+            // We're showing our own UI, ensure the editor doesn't show anything itself.
+            context.OperationContext.TakeOwnership();
+
+            var subjectBuffer = commandArgs.SubjectBuffer;
+            var textView = commandArgs.TextView;
+
+            var caretPoint = textView.GetCaretPoint(subjectBuffer);
+            if (caretPoint is null)
+                return false;
 
+            if (!subjectBuffer.TryGetWorkspace(out var workspace))
+                return false;
+
+            var token = _listener.BeginAsyncOperation(nameof(ExecuteCommand));
+
+            var snapshotSpan = textView.GetTextElementSpan(caretPoint.Value);
+            ExecuteAsync(workspace, commandArgs, snapshotSpan, getCurrentDocumentAsync, getChangedDocumentAsync)
+                .ReportNonFatalErrorAsync()
+                .CompletesAsyncOperation(token);
             return true;
         }
 
-        public bool ExecuteCommand(SortAndRemoveUnnecessaryImportsCommandArgs args, CommandExecutionContext context)
+        private async Task ExecuteAsync(
```

Comment:
❓ What happens if this is called via the command handler while another organize operation is still in progress?

Reply:
the contract for BWIs is that there can be one active per view.  so when another is triggered, the in-flight one is canceled.  (This matches all the other BWIs).  this allowed us to avoid any sort of of complex questions around having multiple showing, or how things like 'esc' should be treated.